Marlin/Marlin_main.cpp

```diff
@@ -5922,18 +5929,17 @@ inline void gcode_M17() {
   }
 
   static void ensure_safe_temperature() {
-    bool did_show = false;
     wait_for_heatup = true;
     while (wait_for_heatup) {
       idle();
+      if (!wait_for_heatup) break;
       wait_for_heatup = false;
       HOTEND_LOOP() {
         if (thermalManager.degTargetHotend(e) && abs(thermalManager.degHotend(e) - thermalManager.degTargetHotend(e)) > 3) {
           wait_for_heatup = true;
-          if (!did_show) { // Show "wait for heating"
+          #if ENABLED(ULTIPANEL)
             lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_WAIT_FOR_NOZZLES_TO_HEAT);
```

Comment:
Looking at this more closely, `lcd_filament_change_show_message` calls `lcd_goto_screen`, which first verifies that `currentScreen != screen` prior to proceeding. As such, I believe it's relative harmless to call `lcd_filament_change_show_message` repeatedly.